### PR TITLE
Closes #2974 Series.map

### LINKED ISSUE #2974: Series.map
Add a Series.map function to match pandas:

https://pandas.pydata.org/docs/reference/api/pandas.Series.map.html

### PR DESCRIPTION
**Pull request by ajpotts**
Closes #2974 Series.map

This adds Series.map function that accepts a dict or arkouda.Series as the mapping correspondence.

### CODE REVIEW

**Comment by ajpotts** [comment]
This may not give the expected results when the np.nan value is involved.  I created a separate tickets:
https://github.com/Bears-R-Us/arkouda/issues/3009
https://github.com/Bears-R-Us/arkouda/issues/3001

**Review by pierce314159 — APPROVED** [review]
This looks great! 2 very small suggestions

File: arkouda/series.py [hunk 1]
```diff
@@ -1008,6 +1008,107 @@ def concat(
...
+        keys = self.values
+        gb = GroupBy(keys, dropna=False)
+        gb_keys = gb.unique_keys
+
+        mapping = arg
+        if isinstance(mapping, dict):
+            mapping = Series([array(list(mapping.keys())), array(list(mapping.values()))])
+
+        if isinstance(mapping, Series):
+            xtra_keys = gb_keys[~in1d(gb_keys, mapping.index.values)]
```
[the remark this answers is not in the document]
Comment (ajpotts) [reply arkouda/series.py]: Good idea; I made the change.

File: arkouda/series.py [hunk 2]
```diff
@@ -1008,6 +1008,107 @@ def concat(
...
+            if xtra_keys.size > 0:
+                nans = akcast(zeros(xtra_keys.size) * np.nan, dt=mapping.values.dtype)
```
[the remark this answers is not in the document]
Comment (ajpotts) [reply arkouda/series.py]: Thanks for pointing this out.  I made the change.

[the remark this answers is not in the document]
Comment (ajpotts) [reply arkouda/series.py]: While I was debugging this I noticed that the Categorical case was not handled, so I added that in as well.

**Review by stress-tess — APPROVED** [review]
Looks good to me! great work!!

**Review by jaketrookman — APPROVED** [review]

**Comment by stress-tess** [comment]
@ajpotts it seems like there are merge conflicts holding this up, once they're resolved i think it's ready to merge

**Review by stress-tess — APPROVED** [review]

**Pull request merged**